src/textual/message_pump.py

```diff
@@ -29,6 +32,32 @@ class MessagePumpClosed(Exception):
     pass
 
 
+T = TypeVar("T")
+
+
+def _is_coroutine_callable(call: Callable) -> bool:
+    if inspect.isroutine(call):
+        return inspect.iscoroutinefunction(call)
+    if inspect.isclass(call):
+        return False
+    call = getattr(call, "__call__", None)
+    return inspect.iscoroutinefunction(call)
+
+
+def _callable_in_thread_pool(call: Callable[..., T]) -> Callable[..., Awaitable[T]]:
+    async def inner(*args: Any, **kwargs: Any) -> T:
+        print(call)
+        loop = asyncio.get_event_loop()
+        # Ensure we run in the same context
+        child = functools.partial(call, *args, **kwargs)
+        context = contextvars.copy_context()
+        func = context.run
+        args = (child,)
+        return await loop.run_in_executor(None, func, *args)
```

Comment:
Given `run_in_executor` schedule the call and returns an awaitable, there seem to be no need to `await` it if you don't make `inner` and async function. 

This would save a round trip to the event loop

Reply:
`*args` is already passed line 52, isn't it redoundant to also pass it line 56? Granted with `partial()` it's not going to matter much, but it shouldn't bring anything either.

Reply:
Not sure I fully understood the latter questions, but I did a bit of simplifying in 9d541a7bcf1389ffec2a9b5e2c0da5a8834f4c77, hopefully that resolves it.

Reply:
> Given `run_in_executor` schedule the call and returns an awaitable, there seem to be no need to `await` it if you don't make `inner` and async function.

Hmm yes, that's a good point! I feel like there might be some cons of returning a regular callable instead of a coroutine function, but I can't immediately think of them, so I'm pushing your suggestion 😄